apps/web/src/app/organizations/manage/group-add-edit.component.ts

```diff
@@ -7,64 +9,144 @@ import { LogService } from "@bitwarden/common/abstractions/log.service";
 import { PlatformUtilsService } from "@bitwarden/common/abstractions/platformUtils.service";
 import { CollectionData } from "@bitwarden/common/models/data/collection.data";
 import { Collection } from "@bitwarden/common/models/domain/collection";
-import { GroupRequest } from "@bitwarden/common/models/request/group.request";
-import { SelectionReadOnlyRequest } from "@bitwarden/common/models/request/selection-read-only.request";
 import { CollectionDetailsResponse } from "@bitwarden/common/models/response/collection.response";
-import { CollectionView } from "@bitwarden/common/models/view/collection.view";
+import { DialogService } from "@bitwarden/components";
+
+import {
+  AccessItemType,
+  AccessItemValue,
+  AccessItemView,
+  convertToPermission,
+  convertToSelectionReadOnly,
+  PermissionMode,
+} from "../components/access-selector";
+import { GroupRequest, GroupServiceAbstraction } from "../services/abstractions/group";
+import { GroupView } from "../views/group.view";
+
+/**
+ * Indices for the available tabs in the dialog
+ */
+export enum GroupAddEditTabType {
+  Info = 0,
+  Members = 1,
+  Collections = 2,
+}
+
+export interface GroupAddEditDialogParams {
+  /**
+   * ID of the organization the group belongs to
+   */
+  organizationId: string;
+
+  /**
+   * Optional ID of the group being modified
+   */
+  groupId?: string;
+
+  /**
+   * Tab to open when the dialog is open.
+   * Defaults to Group Info
+   */
+  initialTab?: GroupAddEditTabType;
+}
 
-import { GroupServiceAbstraction } from "../services/abstractions/group";
+export enum GroupAddEditDialogResultType {
+  Saved = "saved",
+  Canceled = "canceled",
+  Deleted = "deleted",
+}
+
+/**
+ * Strongly typed helper to open a groupAddEditDialog
+ * @param dialogService Instance of the dialog service that will be used to open the dialog
+ * @param config Configuration for the dialog
+ */
+export const openGroupAddEditDialog = (
+  dialogService: DialogService,
+  config: DialogConfig<GroupAddEditDialogParams>
+) => {
+  return dialogService.open<GroupAddEditDialogResultType, GroupAddEditDialogParams>(
+    GroupAddEditComponent,
+    config
+  );
+};
 
 @Component({
   selector: "app-group-add-edit",
   templateUrl: "group-add-edit.component.html",
 })
 export class GroupAddEditComponent implements OnInit {
-  @Input() groupId: string;
-  @Input() organizationId: string;
-  @Output() onSavedGroup = new EventEmitter();
-  @Output() onDeletedGroup = new EventEmitter();
+  protected permissionMode = PermissionMode;
+  protected resultType = GroupAddEditDialogResultType;
```

Comment:
suggestion: use PascalCase is possible
```suggestion
  protected PermissionMode = PermissionMode;
  protected ResultType = GroupAddEditDialogResultType;
```

Reply:
Done